core/shared/src/main/scala/scalaz/zio/Promise.scala

```diff
@@ -135,37 +135,30 @@ object Promise {
    * guarantees that if the resource is acquired (and the state changed), a
    * release action will be called.
    */
-  final def bracket[E, A, B, C](
+  final def bracket[A, B, C](
     ref: Ref[A]
-  )(acquire: (Promise[E, B], A) => (IO[Void, C], A))(release: (C, Promise[E, B]) => IO[Void, Unit]): IO[E, B] =
+  )(
+    acquire: (Promise[Nothing, B], A) => (IO[Nothing, C], A)
```

Comment:
I think we want to leave the `E` in `Promise`, so we'll return `IO[E, B]` here. This will allow the promise to fail. Otherwise, it cannot fail, which will reduce applications for this code.